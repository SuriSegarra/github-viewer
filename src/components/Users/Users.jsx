import React from 'react';
import PropTypes from 'prop-types';
import User from '../User/User';

const Users = ({ users }) => {
  const userElement = users.map(user => (
    //key gives an identifier to react to each of this things
    <li key={user.id}>
      <User {...user} />
    </li>
  ));
  
  return (
    <ul>
      {userElement}
    </ul>
  );
};

Users.propTypes = {
  users: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    login: PropTypes.string.isRequired,
    followers_url: PropTypes.string.isRequired,
    following_url: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    avatar_url: PropTypes.string.isRequired    
  })).isRequired
};

export default Users;
