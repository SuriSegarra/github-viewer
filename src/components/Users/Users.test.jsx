import React from 'react';
import { shallow } from 'enzyme';
import Users from './Users';

describe('Users', () => {
  it('match a users snapshot', () => {
    const users = [
      {
        login: 'Jake',
        followers_url:'1',
        following:'5',
        url:'www.jacob-cline.com',
        avatar_url: 'https://www.fillmurray.com/640/360'
      },
      {
        login: 'Nicholas',
        followers_url:'10',
        following:'59',
        url:'www.nicholascage.com',
        avatar_url: 'https://www.placecage.com/640/360'
      },
      {
        login: 'steven',
        followers_url:'17',
        following:'19',
        url:'www.steven.com',
        avatar_url: 'https://www.stevensegallery.com/640/360'
      },
    ];
    const wrapper = shallow(<Users users={users}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
