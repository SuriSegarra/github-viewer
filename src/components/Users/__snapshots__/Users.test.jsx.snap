// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users match a snapshot 1`] = `
<ul>
  <li>
    <User
      avatar_url="https://www.fillmurray.com/640/360"
      followers_url="1"
      following="5"
      login="Jake"
      url="www.jacob-cline.com"
    />
  </li>
  <li>
    <User
      avatar_url="https://www.placecage.com/640/360"
      followers_url="10"
      following="59"
      login="Nicholas"
      url="www.nicholascage.com"
    />
  </li>
  <li>
    <User
      avatar_url="https://www.stevensegallery.com/640/360"
      followers_url="17"
      following="19"
      login="steven"
      url="www.steven.com"
    />
  </li>
</ul>
`;

exports[`Users match a users snapshot 1`] = `
<ul>
  <li>
    <User
      avatar_url="https://www.fillmurray.com/640/360"
      followers_url="1"
      following="5"
      login="Jake"
      url="www.jacob-cline.com"
    />
  </li>
  <li>
    <User
      avatar_url="https://www.placecage.com/640/360"
      followers_url="10"
      following="59"
      login="Nicholas"
      url="www.nicholascage.com"
    />
  </li>
  <li>
    <User
      avatar_url="https://www.stevensegallery.com/640/360"
      followers_url="17"
      following="19"
      login="steven"
      url="www.steven.com"
    />
  </li>
</ul>
`;
